src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineStartService.java

```diff
@@ -0,0 +1,134 @@
+	 * @throws IridaWorkflowNotFoundException thrown if the workflow cannot be found
+	 * @throws ReferenceFileRequiredException thrown if a reference file is required and not sent (should not happen).
+	 */
+	public Long start(UUID id, LaunchRequest request, Locale locale)
+			throws IridaWorkflowNotFoundException, ReferenceFileRequiredException {
+		IridaWorkflow workflow = workflowsService.getIridaWorkflow(id);
+		IridaWorkflowDescription description = workflow.getWorkflowDescription();
+
+		/*
+		PARAMETERS
+		 */
+		IridaWorkflowNamedParameters namedParameters = null;
+
+		if (request.getSavedParameters() != null) {
+			namedParameters = namedParametersService.read(request.getSavedParameters());
+		}
+
+		/*
+		SHARE RESULTS BACK TO PROJECTS?
+		 */
+		List<Project> projects = new ArrayList<>();
+		if (request.getProjects()
+				.size() > 0) {
+			projects = (List<Project>) projectService.readMultiple(request.getProjects());
+		}
+
+		/*
+		SEQUENCE FILES
+		 */
+		List<SingleEndSequenceFile> singles = new ArrayList<>();
+		List<SequenceFilePair> pairs = new ArrayList<>();
+		// Check for single ended sequence files
+		Iterable<SequencingObject> sequencingObjects = sequencingObjectService.readMultiple(request.getFileIds());
+		if (description.acceptsSingleSequenceFiles()) {
+			sequencingObjects.forEach(sequencingObject -> {
+				if (sequencingObject instanceof SingleEndSequenceFile) {
+					singles.add((SingleEndSequenceFile) sequencingObject);
+				}
+			});
+		} else if (description.acceptsPairedSequenceFiles()) {
+			sequencingObjects.forEach(sequencingObject -> {
+				if (sequencingObject instanceof SequenceFilePair) {
+					pairs.add((SequenceFilePair) sequencingObject);
+				}
+			});
+		}
+
+		// Make sure there is a reference files if one is required.
+
+		if (description.requiresReference() && request.getReference() == null) {
+			throw new ReferenceFileRequiredException(
+					messageSource.getMessage("server.ReferenceFiles.notFound", new Object[] {}, locale));
+		}
+
+		IridaWorkflowInput inputs = description.getInputs();
+		Long submissionId;
+		if (inputs.requiresSingleSample()) {
+			Collection<AnalysisSubmission> submissions = submissionService.createSingleSampleSubmission(workflow,
+					request.getReference(), singles, pairs, request.getParameters(), namedParameters, request.getName(),
+					request.getDescription(), projects, request.isUpdateSamples(), request.isEmailPipelineResult());
+			submissionId = submissions.stream()
+					.findFirst()
+					.orElseThrow()
+					.getId();
+		} else {
+			AnalysisSubmission submission = submissionService.createMultipleSampleSubmission(workflow,
+					request.getReference(), singles, pairs, request.getParameters(), namedParameters, request.getName(),
+					request.getDescription(), projects, request.isUpdateSamples(), request.isEmailPipelineResult());
+			submissionId = submission.getId();
+		}
+		cartService.emptyCart();
```

Comment:
So I came across an issue when running an analysis. This here clears the cart but I'm guessing there is a map or something somewhere that also needs to be cleared.

Here is the issue:

1) Added a sample to the cart
2) Ran the SISTR pipeline
3) The pipeline launches and the cart icon no longer has a badge. When clicking the cart icon there are no samples
4) Went to the project and added the same sample from above to the cart
5) Expected sample to be added to cart but got back the duplicate sample name in cart error

Reply:
This actually has already been fixed.  It is in dev I just need to merge it through.  Holding off for the next PR to pull up dev since your email crap is in it.

Reply:
Lol sounds good!